runners/core-java/src/main/java/org/apache/beam/sdk/util/AssignWindowsDoFn.java

```diff
@@ -64,8 +69,8 @@ public Instant timestamp() {
                 }
 
                 @Override
-                public Collection<? extends BoundedWindow> windows() {
-                  return c.windowingInternals().windows();
+                public BoundedWindow window() {
+                  return Iterables.getOnlyElement(c.windowingInternals().windows());
```

Comment:
future improvement idea: I kind of wish windowingInternals().window() existed, especially since many windowed values have to construct a single element iterable just to have us do this...